accepted/2020/net5/net5.md

```diff
@@ -279,6 +279,73 @@ We'll have the following TFMs:
 |                 | (+everything else inherited from net5.0)                   |                                   |
 | Tizen, Unity... | Will follow the Xamarin model                              |                                   |
 
+### Precedences
+
+We need to decide what precedences we want the new TFMs to have in relation to
+the old ones as well as in relation with the new universally portable TFM
+`net5.0` (and higher).
+
+Let's look at two examples:
+
+* **Example A**
+  * Project targets `net6.0-ios`
+  * Package offers  
+    * `net6.0`
+    * `net5.0-ios`
+* **Example B**
+  * Project targets `net6.0-ios`
+  * Package offers  
+    * `net6.0`
+    * `xamarin.ios`
+
+There are effectively two options:
+
+1. Prefer platform-specific assets, regardless how old their version is.
+2. Prefer the assets that are closer in version number.
+
+Example | Option (1)    | Option (2)
+--------|---------------|--------------
+A       | `net5.0-ios`  | `net6.0`
+B       | `xamarin.ios` | `net6.0`
+
+We prefer Option (2). @mhutch described it best:
+
+> I prefer the Option (2) because it allows re-converging to platform-neutral:
+> in both A and B, if `net6.0` added functionality that meant that the iOS
+> specific implementation was no longer necessary, then the package would not
+> need to have `net6.0-ios` assets.
+>
+> If the `net5.0-ios` assembly has iOS-specific _functionality_, then when the
+> package adds `net6.0` it should also add `net6.0-ios`, else it feels to me
+> like a broken package.
+
+@dsplaisted also added this:
+
+> One of the main benefits of the target framework changes in .NET 5 and on is
+> that it aligns the version numbers of the platform specific targets
+> (`net5.0-ios`) with the portable target (`net5.0`). This makes it easy to
+> understand the version compatibility rules. It also means that we can have a
+> simple and easy understand rule for choosing the best asset that is more
+> likely to produce a result that matches the intent of a package author.
+
+And finally @ericstj added that the current model (where we prefer
+platform-specific assets, regardless of version number) isn't helpful:
+
+> Option (2) makes sense to me. I don't recall a place where we've actually
+> preferred that RIDs had precedence over TFM. Usually it's a problem and
+> results in bloated packages. I like reversing the precedence here.
+
+This leads to these rules:
+
+* Select list of applicable assets
+  * This will include `netX.Y`, `netX.Y-*`, and `xamarin.*` if they are
+    considered compatible
+* If there is an exact match for the project's TFM, pick that
+  * Exact match includes TFI, TFV, TPI and TPV
+* Group `netX.Y*` by X.Y and sort groups from highest to lowest
+* If the first group contains both a `netX.Y` and a `netX.Y-*` or a `xamarin.*`,
```

Comment:
I think this is backwards.  As I read this currently, it would prefer `net5.0` over `net5.0-ios`.

Reply:
> I think this is backwards. As I read this currently, it would prefer `net5.0` over `net5.0-ios`.

That's intended. Rule 2 makes sure that if you're in `net6.0-ios`, it will use `net6.0-ios` if the package offers it. But if the package offers a `net5.0-ios` and a `net6.0`, it will pick `net6.0` based on rule 4.

> 1. Select list of applicable assets
>    * This will include `netX.Y`, `netX.Y-*`, and `xamarin.*` if they are considered compatible
> 2. If there is an exact match for the project's TFM, pick that
>    * Exact match includes TFI, TFV, TPI and TPV
> 3. Group `netX.Y*` by X.Y and sort groups from highest to lowest
> 4. If the first group contains both a `netX.Y` and a `netX.Y-*` or a `xamarin.*`, pick `netX.Y`

Does that make sense?

Reply:
No, I'm not getting it.  You can't rely on just exact matches.  What about this case, for example?

  * Project targets `net6.0-ios`
  * Package offers  
    * `net5.0-ios`
    * `net5.0`

Rule 2 wouldn't apply as it's not an exact match, and rule 4 would mean it would prefer `net5.0`, as far as I can tell.

I think in needs to pick `netX.Y-*` over `netX.Y` as long as they are already compatible.

Reply:
I confused myself. Rule (1) already filters out everything that isn't compatible and rule (2) will group by .NET version. So I think you're correct with your initial suggestion. Fixed.